config/string.c

```diff
@@ -198,24 +198,26 @@ static int string_string_plus_equals(const struct ConfigSet *cs, void *var, cons
 
   int rc = CSR_SUCCESS;
 
-  if (var)
+  char *str;
+  mutt_str_asprintf(&str, "%s%s", mutt_buffer_strdup(var), value);
```

Comment:
Ah, I see what's going on.
`var` is a pointer to the variable.
In this case, the variable is `char *C_Name` and `var` is `char **`
`var` (the pointer) will never be `NULL`, but the variable `*var` _might_ be (variable is empty).

That gives us two cases:
- `*var == NULL` -- just copy the value
- `*var != NULL` -- join the two strings

We've already filtered out the case where `value` is empty.

We don't want `mutt_buffer_strdup()`, just `*var`

Reply:
Good, thanks. I think I sorted it out.